spec/requests/api/v1/subscription_spec.rb

```diff
@@ -0,0 +1,53 @@
+require 'stripe_mock'
+
+RSpec.describe 'User can buy subscritption' do
+  let(:stripe_helper) { StripeMock.create_test_helper }
+    before(:each) { StripeMock.start }
+    after(:each) { StripeMock.stop }
+  let(:user) { create(:user) }
+  let(:user_credentials) { user.create_new_auth_token }
+  let(:headers) { { HTTP_ACCEPT: 'application/json' }.merge!(user_credentials) }
+
+  let(:successful_token) { post '/api/v1/subscriptions', 
+    params: {
+    stripeToken: stripe_helper.generate_card_token    
+    }, headers: headers
+  user.reload }
+
+  let(:invalid_token) { post '/api/v1/subscriptions', 
+    params: {
+    stripeToken: 123456789
+    }, headers: headers}
+
+  describe "User pays for subscription" do
+    before do
+      successful_token
```

Comment:
I see no reason to extract the request to a variable, you guys have not done it previously. Keep it consistent.

Reply:
right you are oliver! i put the request in the before do block